### ISSUE TITLE
netdata display problem with javaFX webview

### ISSUE THREAD

**Issue by limbo127** [firehol/netdata#2103]
Hello, 
we are trying to use netdata in javaFx webview component, but there's some bad display data in graph 
- need to refresh page whereas, graph is displaying bad
- values round does not work

=> Perhaps this issue is relative to webview component . 

javaFx webview is based on : 
 https://docs.oracle.com/javase/8/javafx/api/javafx/scene/web/WebView.html
Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/602.1 (KHTML, like Gecko) JavaFX/8.0 Safari/602.1

**Comment by limbo127** [comment]
javaFx example : 
import javafx.application.*;
import javafx.scene.*;
import javafx.scene.layout.*;
import javafx.scene.web.*;
import javafx.stage.*;

public class WebViewTest extends Application {

    @Override
    public void start(Stage stage) {
         String url="https://aws-fra.my-netdata.io/default.html";
        stage.setTitle(url);
        Scene scene = new Scene(new Group());

        BorderPane root = new BorderPane();

        final WebView browser = new WebView();
        final WebEngine webEngine = browser.getEngine();
        browser.setPrefWidth(1024);
            browser.setPrefHeight(768);
        webEngine.load(url);

        root.setCenter(browser);
        scene.setRoot(root);

        stage.setScene(scene);
        stage.show();
    }

    public static void main(String[] args) {
        launch(args);
    }
}

**Comment by limbo127** [comment]
screenshot 
![javafx-webview](https://cloud.githubusercontent.com/assets/1582800/25180793/ad37484c-250f-11e7-8fef-1c259bb3d782.png)

**Comment by ktsaou** [comment]
Hi,

I see 2 problems:

1. dygraphs are not working properly (stretched charts)
2. numbers have way too many decimal points.

I can fix No 2.
I am not sure if No 1 is a netdata problem or a dygraphs problem. Could you please check if plain dygraphs are working? They have a lot of examples at their site: http://dygraphs.com/

**Comment by limbo127** [comment]
thanks, 
after test,  issue comes from dygraph for point 1, i submit a bug report
Regards, 
NIcolas

**Comment by ktsaou** [comment]
Regarding issue No 2, I found out Safari 8 has the same issue.

Netdata does [this](https://github.com/firehol/netdata/blob/0fa54a088b160d4d17348c3cacbbd90f36ecb346/web/dashboard.js#L2302-L2310):

```js
            return value.toLocaleString(undefined, {
                useGrouping: true,
                minimumFractionDigits: dmin,
                maximumFractionDigits: dmax
            });
```

I traced it down to this:

```
dmin  1 , dmax  1 , value  2.48756
dmin  1 , dmax  1 , value  4.97512
```

So, netdata requests 1 decimal digits, but `toLocaleString()` returns 5.

I researching alternatives...

**Comment by ktsaou** [comment]
ok, I think I did it (for the problem No 2).

netdata now has 3 implementation for formating values:

1. Using `Intl.NumberFormat()` - I just added this, it is the fastest of all.
2. Using `Number.toLocaleString()` - the one used before, which gives wrong decimal points on older systems (like Safari 8, and I guess javaFX webview).
3. Using `Number.toFixed()` - which should work everywhere, but it is not locale aware. That is, no thousands seperators.

netdata will prefer them in the above order. It tests them to find if they work as expected (i.e. providing the right number of fractional digits) or not.
